### PR TITLE
improve log smileys

### LINKED ISSUE #12471: Add text and emoji for log smileys
### Tell us your idea!

To make selecting the right smiley easier show also the description.

### What solution would you suggest?

Instead:
Screenshot_20220104-214354_cgeo

You will have:
grafik

### Is there an alternative?

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by bekuno**
fixes #12471

- add smiley description to the smiley
- add translation possibility
- add and fix OC smileys

### CODE REVIEW

**Comment by bekuno** [comment]
@Lineflyer (or other)
Can you check from c:geo, if the smiley on OCDE side are accepted?
It is not possible from the web dialog.

File: main/res/values-de/strings.xml [hunk 1]
```diff
@@ -2163,4 +2163,40 @@
     <!-- ChipChoiceGroup -->
     <string name="chipchoicegroup_selectall">Alles auswählen</string>
     <string name="touch_again_to_exit">Zum Beenden erneut tippen</string>
+
+    <!-- Smileys GC / OC -->
+    <string name="smiley_smile">Lächeln</string>
+    <string name="smiley_big_smile">Fettes Lächeln</string>
+    <string name="smiley_cool">Kalt</string>
```
Comment (ztNFny) [inline main/res/values-de/strings.xml:2170]: "Cool"

File: main/res/values-de/strings.xml [hunk 2]
```diff
@@ -2163,4 +2163,40 @@
...
+    <string name="smiley_blush">Erröten</string>
+    <string name="smiley_tongue">Zunge</string>
+    <string name="smiley_evil">Teufel</string>
+    <string name="smiley_wink">Winken</string>
```
Comment (ztNFny) [inline main/res/values-de/strings.xml:2174]: "Zwinkern"

File: main/res/values-de/strings.xml [hunk 3]
```diff
@@ -2163,4 +2163,40 @@
...
+    <string name="smiley_clown">Clown</string>
+    <string name="smiley_blackeye">Schwarze Augen</string>
```
Comment (ztNFny) [inline main/res/values-de/strings.xml:2176]: "Blaues Auge"

File: main/res/values-de/strings.xml [hunk 4]
```diff
@@ -2163,4 +2163,40 @@
     <!-- ChipChoiceGroup -->
     <string name="chipchoicegroup_selectall">Alles auswählen</string>
     <string name="touch_again_to_exit">Zum Beenden erneut tippen</string>
+
+    <!-- Smileys GC / OC -->
+    <string name="smiley_smile">Lächeln</string>
+    <string name="smiley_big_smile">Fettes Lächeln</string>
```
Comment (ztNFny) [inline main/res/values-de/strings.xml:2169]: "Lachen"

File: main/res/values-de/strings.xml [hunk 5]
```diff
@@ -2163,4 +2163,40 @@
...
+    <string name="smiley_cool">Kalt</string>
+    <string name="smiley_blush">Erröten</string>
+    <string name="smiley_tongue">Zunge</string>
```
Comment (ztNFny) [inline main/res/values-de/strings.xml:2172]: "Zunge rausstrecken"

File: main/res/values-de/strings.xml [hunk 6]
```diff
@@ -2163,4 +2163,40 @@
...
+    <string name="smiley_evil">Teufel</string>
+    <string name="smiley_wink">Winken</string>
+    <string name="smiley_clown">Clown</string>
+    <string name="smiley_blackeye">Schwarze Augen</string>
+    <string name="smiley_8ball">8er Ball</string>
+    <string name="smiley_sad">Traurig</string>
+    <string name="smiley_shy">Schüchtern</string>
+    <string name="smiley_shocked">Schockiert</string>
+    <string name="smiley_angry">Ärgerlich</string>
```
Comment (ztNFny) [inline main/res/values-de/strings.xml:2181]: "Böse"

File: main/res/values-de/strings.xml [hunk 7]
```diff
@@ -2163,4 +2163,40 @@
...
+    <string name="smiley_dead">Tot</string>
+    <string name="smiley_sleepy">Schlafend</string>
+    <string name="smiley_kisses">Küsse</string>
```
Comment (ztNFny) [inline main/res/values-de/strings.xml:2184]: "Kuss"

File: main/res/values-de/strings.xml [hunk 8]
```diff
@@ -2163,4 +2163,40 @@
...
+    <string name="smiley_approve">Zustimmung</string>
+    <string name="smiley_disapprove">Ablehnung</string>
+    <string name="smiley_question">Frage</string>
+    <string name="smiley_cry">Schrei</string>
```
Comment (ztNFny) [inline main/res/values-de/strings.xml:2188]: "Weinen"

**Review by ztNFny — COMMENTED** [review]
suggestions for german strings

**Comment by moving-bits** [comment]
FYI: No need to include German (or any other) translations at all - they will be ignored and overwritten by Crowdin anyway.

File: main/res/values/strings.xml [hunk 9]
```diff
@@ -2427,4 +2427,39 @@
...
+    <string name="smiley_innocent">innocent</string>
+    <string name="smiley_kiss">kiss</string>
+    <string name="smiley_laughing">laughing</string>
+    <string name="smiley_money">money</string>
+    <string name="smiley_sealed">sealed</string>
+    <string name="smiley_surprised">surprised</string>
+    <string name="smiley_tongueout">tongue out</string>
+    <string name="smiley_undecided">undecided</string>
+    <string name="smiley_yell">yell</string>
+
```
Comment (Lineflyer) [inline main/res/values/strings.xml:2464]: Not being an expert, but shouldn't such kind of terms start with a capital letter in that single word use case?

**Comment by bekuno** [comment]
rebased to master, but got an error on CI :
~~~
/home/jenkins/slave/workspace/cgeo-CI_PR-build/main/src/cgeo/geocaching/maps/mapsforge/v6/RenderThemeSettingsFragment.java:92: error: cannot find symbol
            Settings.setSelectedMapRenderThemeStyle(preference.getKey(), newValue.toString());
            ^
  symbol:   variable Settings
  location: class RenderThemeSettingsFragment
~~~

**Comment by moving-bits** [comment]
> rebased to master, but got an error on CI :

see #12478, fixed meanwhile on `master`

**Comment by bekuno** [comment]
Thanks for all ideas and support.
The german strings where only created to demonstrate the translation. The language file is planned to delete when the PR is ready. I am open to better wording, the direct translations where not good. But I think we should avoid slang words - the "cool" is imho not fine.
@Lineflyer You are our translation expert. It looks better starting the text with capital letter.

This is the next mockup:
grafik

What do you mean to - is this so ok?

**Comment by ztNFny** [comment]
Cool slang? 🤣

https://www.duden.de/rechtschreibung/cool

**Comment by eddiemuc** [comment]
If in doubt wrt how to name the emojis, I suggest to fall back to the names given by GC.com itself:
https://www.geocaching.com/help/index.php?pg=kb.chapter&id=107&pgid=739

**Comment by Lineflyer** [comment]
> @Lineflyer You are our translation expert. It looks better starting the text with capital letter.

I love this capital letter discussion ;)
I recently had the same issue when providing translation for the map legend on Crowdin.
AFAIK its not mandatory to start with capital letter in such case, but there are some factors, which suggest it:
- Start of a sentence has to be capitalized
- Fixed terms can be capitalized

So IMHO yes, it should be capitalized as it just looks more nice (at least for my german eyes).

**Comment by bekuno** [comment]
> Cool slang? 🤣
> 
> https://www.duden.de/rechtschreibung/cool

"Gebrauch: salopp"
This is not "umgangssprachlich", wich I would prefer.

**Comment by bekuno** [comment]
@teiling88
My test result minutes before shows, that (at least using the website) OCDE does no more support plain smileys. 
Also the log site java script `function insertSmiley(smileySymbol, smileyPath)` does not use the `smileySymbol` entry.

If OCDE (or OC in general) does no more support plain smileys, then I would remove it within this PR.

**Comment by Lineflyer** [comment]
IMHO there have been plans by OC(.de) to support similar syntax for links, smilies, etc. as geocaching.com does. Not sure though.

**Comment by teiling88** [comment]
Did it work in a earlier stage? We didn't change any thing in the smilie handling as far as ! know.

**Comment by bekuno** [comment]
> Did it work in a earlier stage? We didn't change any thing in the smilie handling as far as I know.

@teiling88 
Don´t know.
The implementation was created by @Bananeweizen in october 2016 in b0b421f59d58f46c613a8a5b141ae8a2a4155f3f .

Btw., how can I file an issue for https://github.com/OpencachingDeutschland/oc-server3 ?
The redmine link does not work anymore.

**Comment by bekuno** [comment]
> > Did it work in a earlier stage? We didn't change any thing in the smilie handling as far as I know.
> 
> @teiling88 Don´t know. The implementation was created by @Bananeweizen in october 2016 in [b0b421f](https://github.com/cgeo/cgeo/commit/b0b421f59d58f46c613a8a5b141ae8a2a4155f3f) .
> 
> Btw., how can I file an issue for https://github.com/OpencachingDeutschland/oc-server3 ? The redmine link does not work anymore.


@teiling88  I found the link to jira.
Maybe the non more workin text emoji are a result of working on this ticket: https://opencaching.atlassian.net/browse/RED-1225 ?

**Comment by bekuno** [comment]
I fixed now the missing emoji by using the emoji support library.
Now we have
API21:
2022-02-28 22_09_09-cgeo – abstract_logging_activity xml  2022-02-28 22_09_49-cgeo – abstract_logging_activity xml

API30:
2022-02-28 22_08_19-cgeo – abstract_logging_activity xml  2022-02-28 22_08_40-cgeo – abstract_logging_activity xml

Are the Icons ok or are there other suggestions?

**Comment by bekuno** [comment]
I will remove the Emoji support for OC, as there is no such implementation.

**Comment by bekuno** [comment]
retest this please

**Comment by bekuno** [comment]
retest this please

**Comment by bekuno** [comment]
@mucek4 / @Lineflyer 
The executor cgeo-1 is in state "Could not create ADB Bridge".
Please rebote them.
I deactivated cgeo-1 for now.

**Comment by eddiemuc** [comment]
> Are the Icons ok or are there other suggestions?

I like them. I vote for merging. If discussions start later for single icons then a separate PR would be easy to do. Thanks for the contribution!

**Review by Lineflyer — APPROVED** [review]
I do also like them.

**Comment by bekuno** [comment]
> If OCDE (or OC in general) does no more support plain smileys, then I would remove it within this PR.

@teiling88 FYI - I deactivated now the log smiley part for OC caches.

**Pull request merged**